File: Release/WetiMatlabFunctions/Processing/ProcessingSimulations.m
function [status,result] = ProcessingSimulations(SimulationFolder,SimulationNames,ExeFile,Tool,options)

% inputs
arguments
    SimulationFolder    char       
    SimulationNames     cell
    ExeFile             char
    Tool                {mustBeMember(Tool,{'OpenFAST','Flex5'})} = 'OpenFAST'
    options.nCore       {mustBeInteger} = maxNumCompThreads % set option to 0 for no parallel processing
end

% get dimensions and allocation
n_Simulation    = size(SimulationNames,1);
result          = cell(1,n_Simulation);
status          = NaN(1,n_Simulation);

% move to SimulationFolder
MainFolder      = pwd;
cd(SimulationFolder)

% run simulations
switch Tool
    case 'Flex5'
        for i_Simulation    = 1:n_Simulation   
            SimulationName  = SimulationNames{i_Simulation};
            fprintf('Running simulation %s (%d/%d)\n',SimulationName,i_Simulation,n_Simulation);        
            [status(i_Simulation),result{i_Simulation}] = ...
                system([ExeFile,...
                ' ',SimulationName,'.inf',...
                ' ',SimulationName,'.log',...
                ' ',SimulationName,'.res']);
        end
    case 'OpenFAST'
        parfor (i_Simulation    = 1:n_Simulation,options.nCore)    
            SimulationName  = SimulationNames{i_Simulation};
            fprintf('Running simulation %s (%d/%d)\n',SimulationName,i_Simulation,n_Simulation);        
            [status(i_Simulation),result{i_Simulation}] = ...
                system([ExeFile,...
                ' ',SimulationName,'.fst']);
        end
end

% back to main folder
cd(MainFolder)

end

File: Release/PythonFunctions/Processing/ProcessingSimulations.py
# Setup
import os
from joblib import Parallel, delayed
from subprocess import getstatusoutput
from multiprocessing import cpu_count


def ProcessingSimulations(SimulationFolder, SimulationNames, ExeFile, Tool='OpenFAST', nCore=None):
    if nCore is None:
        nCore = cpu_count()

    # get dimensions and allocation
    n_Simulation = len(SimulationNames)
    result = [None] * n_Simulation
    status = [float('nan')] * n_Simulation

    # move to SimulationFolder
    MainFolder = os.getcwd()
    os.chdir(SimulationFolder)

    def run_simulation(i_Simulation):
        SimulationName = SimulationNames[i_Simulation]
        print(f"Running simulation {SimulationName} ({i_Simulation + 1}/{n_Simulation})")
        if Tool == 'Flex5':
            command = f"{ExeFile} {SimulationName}.inf {SimulationName}.log {SimulationName}.res"
        elif Tool == 'OpenFAST':
            command = f"{ExeFile} {SimulationName}.fst"
        return getstatusoutput(command)

    # run simulations
    results = Parallel(n_jobs=nCore)(delayed(run_simulation)(i) for i in range(n_Simulation))

    for i, (status_code, output) in enumerate(results):
        status[i] = status_code
        result[i] = output

    # back to main folder
    os.chdir(MainFolder)

    return status, result
